### PR TITLE
Stlx 14831 withdraw builder

### PR DESCRIPTION
**Pull request by facundoCarreSimtlix**
Add the withdraw builder of Near at account-lib

This is the ticket : https://bitgoinc.atlassian.net/browse/STLX-14831

### CODE REVIEW

File: modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { BuildTransactionError } from '../baseCoin/errors';
+
+export class StakingWithdrawBuilder extends TransactionBuilder {
+  private contractCallWrapper;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this.contractCallWrapper = new ContractCallWrapper();
+    this.contractCallWrapper.methodName = StakingContractMethodNames.Withdraw;
+    this.contractCallWrapper.deposit = '0'; //
```
Comment (ipelaez-simtlix) [inline modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts:21]: delete that // , lets delete also same thing on Deactivate Builder

**Review by ipelaez-simtlix — CHANGES_REQUESTED** [review]
Lets use stakingActivate as reference and add checks to intput and output on some unit tests here.

File: modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import assert from 'assert';
+import BN from 'bn.js';
+
+import { ContractCallWrapper } from './contractCallWrapper';
+import { TransactionBuilder } from './transactionBuilder';
+import { StakingContractMethodNames } from './constants';
+import { BuildTransactionError } from '../baseCoin/errors';
+
+export class StakingWithdrawBuilder extends TransactionBuilder {
+  private contractCallWrapper;
```
Comment (silalvado) [inline modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts:15]: type is missing

File: modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      const tx = await txBuilder.build();
+      tx.inputs.length.should.equal(1);
+      should.equal(tx.type, TransactionType.StakingWithdraw);
+      const rawTx = tx.toBroadcastFormat();
+      should.equal(rawTx, testData.rawTx.stakingWithdraw.unsigned);
+    });
+
+    it('should fail to staking withdraw with missing gas', async () => {
```
Comment (silalvado) [inline modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts:64]: This shouldn't be inside succeed describe

File: modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts (hunk 2 above)
Comment (silalvado) [inline modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts:15]: Type is missing

File: modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      txBuilder
+        .amount(amount)
+        .sender(testData.accounts.account1.address, testData.accounts.account1.publicKey)
+        .receiverId(validatorContractAddress)
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      await txBuilder.build().should.be.rejectedWith('gas is required before building staking withdraw');
+    });
+
+    it('should fail to staking withdraw with missing amount', async () => {
```
Comment (silalvado) [inline modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts:75]: This shouldn't be inside succeed describe

File: modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            methodName: 'withdraw',
+            args: { amount: amount },
+            gas: '125000000000000',
+            deposit: '0',
+          },
+        },
+      ]);
+    });
+
+    it('build from an signed staking withdraw', async () => {
```
Comment (silalvado) [inline modules/account-lib/test/unit/coin/near/transactionBuilder/stakingWithdrawBuilder.ts:108]: replace an with a

File: modules/account-lib/src/coin/near/transaction.ts [hunk 6]
```diff
@@ -224,6 +227,20 @@ export class Transaction extends BaseTransaction {
           coin: this._coinConfig.name,
         });
         break;
+      case TransactionType.StakingWithdraw:
+        const unit8ArrayToString = Buffer.from(action.functionCall.args).toString();
```
Comment (silalvado) [inline modules/account-lib/src/coin/near/transaction.ts:231]: unnecessary constant

**Review by silalvado — CHANGES_REQUESTED** [review]

**Review by ipelaez-simtlix — APPROVED** [review]
I have added some code here but I had request changes before so I need approve to unock this PR

File: modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  public gas(gas: string): this {
+    this.validateValue(new BigNumber(gas));
+    this.contractCallWrapper.gas = gas;
+    return this;
+  }
+
+  /**
+   * Sets the amount of this transaction.
+   *
+   * @param {string} value the amount in the minimum unit (yocto) of this transaction
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/near/stakingWithdrawBuilder.ts:55]: specifgy here number of decimals.

**Review by silalvado — APPROVED** [review]

**Review by rliberal-simtlix — APPROVED** [review]

**Pull request merged**